src/components/Forms/AddProductForm.tsx

```diff
@@ -67,10 +91,23 @@ const AddProductForm = () => {
 		setValue('images', imageFiles);
 	}, [imageFiles, setValue]);
 
-	const onSubmit: SubmitHandler<productTypes> = async (data: productTypes) => {
+	const onSubmit: SubmitHandler<productTypes> = async (
+		formData: productTypes,
+	) => {
 		try {
-			const res = await dispatch(addProduct(data)).unwrap();
-			showSuccessMessage(res.message);
+			if (productData.id) {
+				setLoading(true);
+				const res = await dispatch(
+					updateProduct({ productData: formData, id: productData.id }),
+				).unwrap();
+				showSuccessMessage(res.message);
+				setLoading(false);
```

Comment:
it may be better if you use try()catch()finally() to avoid the duplication of setLoading(false)